src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowAnalysisResult.cs

```diff
@@ -135,6 +135,19 @@ protected DataFlowAnalysisResult(DataFlowAnalysisResult<TBlockAnalysisResult, TA
             return null;
         }
 
+        internal DataFlowAnalysisResult<TBlockAnalysisResult, TAbstractAnalysisValue>? TryGetInterproceduralResultByDefinition(IMethodSymbol symbol)
+        {
+            foreach (var kvp in _interproceduralResultsMap)
+            {
+                if (kvp.Key is IInvocationOperation iOperation && iOperation.TargetMethod.OriginalDefinition.Equals(symbol))
+                {
+                    return (DataFlowAnalysisResult<TBlockAnalysisResult, TAbstractAnalysisValue>)kvp.Value;
```

Comment:
I think we should return an `IEnumerable` result here and caller should process all results.
```suggestion
                    yield return (DataFlowAnalysisResult<TBlockAnalysisResult, TAbstractAnalysisValue>)kvp.Value;
```

Reply:
Are you sure? it doesn't seem to return IEnumerable, the declaration: ` ImmutableDictionary<IOperation, IDataFlowAnalysisResult<TAbstractAnalysisValue>> _interproceduralResultsMap;` and similar function above not returning IEnumerable
```csharp
    internal DataFlowAnalysisResult<TBlockAnalysisResult, TAbstractAnalysisValue>? TryGetInterproceduralResult(IOperation operation)
    {
        if (_interproceduralResultsMap.TryGetValue(operation, out var result))
        {
            return (DataFlowAnalysisResult<TBlockAnalysisResult, TAbstractAnalysisValue>)result;
        }

        return null;
    }
```

Reply:
The other function you mentioned has a very different purpose. What you are attempting here is identifying the state within a local function invoked from all possible contexts. Can you add a test such as below to confirm?

```csharp
void M()
{
    if (RuntimeInformationHelper.IsOSPlatformOrLater(OSPlatform.Windows, 10))
    {
        LocalM();
    }

    LocalM();
    return;

    void LocalM()
    {
        WindowsOnlyMethod(); // This should be flagged as second invocation of 'LocalM()' has no guard
    }
}

[MinimumOSPlatform("Windows")]
void WindowsOnlyMethod()
{
}
```

Reply:
I see what you meant, we do not need to diagnose the local function invocation, because the local function has no platform attribute and not make sense to have one. In your example, we should warn in the local function definition where the platform-specific method is invoked which is working well with the current implementation, i wrote below test and it run successfully:
```csharp
    void M()
    {
        if (OperatingSystemHelper.IsOSPlatformVersionAtLeast("Windows", 10, 2))
        {
            LocalM();  // We don't need to account this platform check
        }

        LocalM(); // Do not warn here
        return;

        void LocalM()
        {
            if (OperatingSystemHelper.IsOSPlatformVersionAtLeast("Windows", 10, 2))
            {
                WindowsOnlyMethod();
            }
            else
            {
                [|WindowsOnlyMethod()|]; // warns here
            }

            if (OperatingSystemHelper.IsWindows() && !OperatingSystemHelper.IsWindowsVersionAtLeast(10,0))
            {
                UnsupportedWindows10();
            }
            else
            {
                [|UnsupportedWindows10()|]; // and here
            }
        }
    }

    [SupportedOSPlatform("Windows10.1.2.3")]
    public void WindowsOnlyMethod()
    {
    }

    [UnsupportedOSPlatform("Windows10.0")]
    public void UnsupportedWindows10()
    {
    }
```

Reply:
How about below test? I think with current implementation it will not flag it because first invocation to local function is guarded. It should be flagged because second invocation is unguarded.

```csharp
void M()
    {
        if (OperatingSystemHelper.IsOSPlatformVersionAtLeast("Windows", 10, 2))
        {
            LocalM();
        }

        LocalM();
        return;

        void LocalM()
        {
                [|WindowsOnlyMethod()|]; // should warn here
        }
    }

    [SupportedOSPlatform("Windows10.1.2.3")]
    public void WindowsOnlyMethod()
    {
    }
```

Reply:
yes that warns, that is what i checked first, then added platform check to that version, will add that separate call too

Reply:
Actually it is not warning sometimes, now i see your point more clearly, it is actually depending on where the local function called, its not warn for the case local is called within guard method.

So I need to check all results, but i prefer to check that here and return one result. Now i wonder how to check that correctly before i was checking the symbol, but it is the same for both calls, so it was getting whichever came first int the list. Now i have added parent check, it seems working, is that enough? What do you think?
```csharp
    if (kvp.Key is IInvocationOperation iOperation && iOperation.TargetMethod.Equals(symbol) &&
        iOperation.TargetMethod.ContainingType.Equals(symbol.ContainingType))
    {
        return (DataFlowAnalysisResult<TBlockAnalysisResult, TAbstractAnalysisValue>)kvp.Value;
    }
```

Reply:
Checking ContainingSymbol didn't help, looked up for while, seems there is nothing to distinguish calling context from local method definition